src/data/nat/choose.lean

```diff
@@ -16,6 +15,46 @@ have h₃ : ¬p ∣ fact (p - k), from mt hp.dvd_fact.1 (not_le_of_gt (nat.sub_l
 by rw [← choose_mul_fact_mul_fact (le_of_lt hkp), mul_assoc, hp.dvd_mul, hp.dvd_mul] at h₁;
   exact h₁.resolve_right (not_or_distrib.2 ⟨h₂, h₃⟩)
 
+/-- Show that choose is increasing for small values of the right argument. -/
+lemma choose_le_succ_of_lt_half_left {r n : ℕ} (h : r < n/2) :
+  choose n r ≤ choose n (r+1) :=
+begin
+  refine le_of_mul_le_mul_right _ (nat.lt_sub_left_of_add_lt (lt_of_lt_of_le h (nat.div_le_self n 2))),
+  rw ← choose_succ_right_eq,
+  apply nat.mul_le_mul_left,
+  rw [← nat.lt_iff_add_one_le, nat.lt_sub_left_iff_add_lt, ← mul_two],
+  exact lt_of_lt_of_le (mul_lt_mul_of_pos_right h zero_lt_two) (nat.div_mul_le_self n 2),
+end
+
+/--
+Show that for small values of the right argument, the middle value is largest.
+You probably want `choose_le_middle` instead of this.
+-/
+lemma choose_le_middle_of_le_half_left {n r : ℕ} (hr : r ≤ n/2) :
+  choose n r ≤ choose n (n/2) :=
+decreasing_induction
+  (λ _ k a,
+      (eq_or_lt_of_le a).elim
+        (λ t, t.symm ▸ le_refl _)
+        (λ h, trans (dominate_choose_lt h) (k h)))
+  hr (λ _, le_refl _) hr
```

Comment:
Is there any reason to include this?

Reply:
Oh, I did something stupid when renaming these lemmas. This one is used to prove `choose_le_middle` - I know I could put the proof inside that one, but I think this is easier to read.

Reply:
Hopefully this is resolved?